### PR TITLE
Addition of material whitelist/blacklist

### PR DESCRIPTION
**Pull request by TylerS1066**
Another incompatibility I found is that when using the Movecraft plugin, this plugin will detect explosions on moving crafts and regenerate those blocks after the craft has moved or sunk.  The easiest method to resolve this is to add support for a whitelist/blacklist system such that server owners can add the materials on their crafts to the blacklist, or setup a whitelist.

While I was at it, I also implemented tag support when loading the materials so that things like `"#logs"` can be used.

### CODE REVIEW

**Comment by TylerS1066** [comment]
As a note, I put this as a draft until #15 is merged, since I wasn't smart and both features are built on top of each other.

**Comment by pmdevita** [comment]
Yeah, this would probably be a useful feature for a few different cases so thanks for adding it. There's a couple thoughts I have about the implementation so I'll add some comments there.

If you'd like, we can add some integration between the plugins, that would make for a more polished result. We could get the hitbox of the explosion, check if any of the crafts are involved with that, and remove those blocks from CH's handling. If you want to do it on your side, I can add a custom event that CH fires before it starts explosion processing for you to edit it's block list. That should work as long as the event isn't deferred until after the explosion event completes. Alternatively, you could open public access to CraftManager and then CH could iterate through the crafts to check them.

File: src/main/kotlin/net/pdevita/creeperheal2/CreeperHeal2.kt [hunk 1]
```diff
@@ -40,8 +40,16 @@ class CreeperHeal2 : JavaPlugin() {
 //            debugLogger("Explosion with no blocks")
             return null
         }
-//        explosions.add(Explosion(this, blockList))
-        val newExplosion = Explosion(this, blockList)
+
```
[the remark this answers is not in the document]
Comment (pmdevita) [reply src/main/kotlin/net/pdevita/creeperheal2/CreeperHeal2.kt:43]: On second thought, I don't think we can actually reject blocks from an explosion this way, I'll come back with a final answer later

**Review by pmdevita — COMMENTED** [review]
First review

**Pull request closed without merging**